pymc_experimental/dp/dp.py

```diff
@@ -0,0 +1,83 @@
+#   Copyright 2020 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy as np
+import pymc as pm
+import pytensor.tensor as pt
+from pymc.model import modelcontext
+
+__all__ = ["DirichletProcess"]
+
+
+def DirichletProcess(name, alpha, base_dist, K, observed=None, sbw_name=None, atoms_name=None):
+    r"""
+    Truncated Dirichlet Process for Bayesian Nonparametric Density Modelling
+
+    Parameters
+    ----------
+    alpha: tensor_like of float
+        Scale concentration parameter (alpha > 0) specifying the size of "sticks", or generated
+        weights, from the stick-breaking process. Ideally, alpha should have a prior and not be
+        a fixed constant.
+    base_dist: single batched distribution
```

Comment:
Would it make sense to use the same API as in other distribution factories, where the user passes a `.dist` variable and we resize it ourselves (and in this case, register in the model as well)?

Reply:
Yes, good idea